### ISSUE TITLE
Add clause for string maximum and minimum length

### ISSUE THREAD

**Issue by johanvergeer** [feO2x/Light.GuardClauses#62]
Would it be possible to add 2 more guardclauses?

One for the minimum length and one for the maximum length of a string. 

Right now I'm doing it this way:

```cs
MyString.Length.MustBeGreaterThan(MinimumLength);
MyString.Length.MustBeLessThan(MaximumLength);
```

Some suggestions

```cs
String.MustBeLongerThan(MinimumLength)
String.MustHaveSizeOrBeLongerThan(MinimumLength)
String.MustBeEmptyOrBeLongerThen(MinimumLength)

String.MustBeShorterThen(MaximumLength)
String.MustHaveSizeOrBeShorterThen(MaximumLength)
String.MustBeEmptyOrBeShorterThen(MaximumLength)
```

**Comment by feO2x** [comment]
That sounds nice! I like the "longer" and "shorter" versions best. I think, we should implement the following assertions:

```csharp
// Minimum length
string.MustBeLongerThan(minimumLength);
string.MustBeLongerThanOrEqualTo(minimumLength);

// Exact length
string.MustHaveLength(length)

// Maximum length
string.MustBeShorterThan(maximumLength)
string.MustBeShorterThanOrEqualTo(maximumLength)

// Using Range<T>
string.MustHaveLengthIn(Range.FromInclusive(10).ToExclusive(20));
```

Do you want to implement them or should we split them between us?

**Comment by johanvergeer** [comment]
Shall I make an initial setup that you can review? I think that worked last
time.

Op ma 4 mrt. 2019 18:11 schreef Kenny Pflug <notifications@github.com>:

**Comment by feO2x** [comment]
👍 sounds awesome! Thanks for your efforts.

**Comment by feO2x** [comment]
@johanvergeer Hey, it's been some time since we last discussed this issue. How are you progressing? I have some spare time next week (vacations 😄) so I could help with some of the assertions.

**Comment by johanvergeer** [comment]
Hi @feO2x. Sorry, I didn't have any time yet to implement it. I've been working on my thesis like crazy. 

So if you wouldn't mind, you can implement it.

**Comment by feO2x** [comment]
Alright, I reassigned the issue.

**Comment by feO2x** [comment]
Implemented in d9846b645d11dbb5ff0a2d2ea594eaae5d95767f

I will release this with version 7.0 of Light.GuardClauses. This will probably happen in two weeks - I first want to [add assertions for `Span<T>` and `Memory<T>`](https://github.com/feO2x/Light.GuardClauses/issues/63).
